src/pymovements/events/base.py

```diff
@@ -0,0 +1,92 @@
+"""
+This module holds all the main Event classes used for event detection.
+"""
+
+
+class Event:
+    """
+    Base Event class.
+
+    Parameters
+    ----------
+    name: str
+        Name of event.
+    onset: int
+        Starting index of event (included).
+    offset: int
+        Ending index of event (excluded).
+
+    Attributes
+    ----------
+    name: str
+        Name of event.
+    onset: int
+        Starting index of event (included).
+    offset: int
+        Ending index of event (excluded).
+
+    """
+    def __init__(self, name: str, onset: int, offset: int):
+        self.name = name
+        self.onset = onset
+        self.offset = offset
+
+    @property
+    def duration(self) -> int:
+        """
+        Get sample duration of event.
+
+        Returns
+        -------
+        int
+            duration in samples.
+        """
+        return self.offset - self.onset
+
+
+class Fixation(Event):
+    """
+    Fixation class.
+
+    Parameters
+    ----------
+    onset: int
+        Starting index of event (included).
+    offset: int
+        Ending index of event (excluded).
+    position: tuple[float]
+        (x, y) position of fixation
+    """
+    _name = 'fixation'
+
+    def __init__(self, onset: int, offset: int, position: tuple[float]):
```

Comment:
change `position` type hint to `tuple[float, float]`
right now this won't show as an error because we don't have mypy integrated yet.

but just  for clarification:
- `tuple[float]` refers to a tuple of length 1 with a single `float` like `(1, )`
- `tuple[float, float]` refers to a tuple of length 2 with both elements `float` like `(1, 1)`
- `tuple[float, str]` refers to a tuple of length 2 with a `float` and a `str` like `(1, 'abc')`
- `tuple[float, ...]` refers to a tuple of variable length like `(1, 1, 1, 1, 1)`

see for reference:
https://docs.python.org/3/library/typing.html#typing.Tuple